src/main/java/org/mifos/connector/mpesa/camel/routes/OperationsRoute.java

```diff
@@ -24,45 +32,81 @@
+                    exchange.getIn().setBody(""+isRe);
+                });
 
         from("direct:filter-by-error-code")
                 .id("filter-by-error-codes")
                 .log(LoggingLevel.INFO, "### Starting FILTER-BY-ERROR-CODE route")
-                .toD(getFilterUrl(FILTER_BY_ERROR_CODE, exchangeProperty(OPERATIONS_FILTER_VALUE)))
+                .removeHeader("*")
+                .removeHeader("Authorization")
+                .setHeader(Exchange.HTTP_METHOD, constant("GET"))
+                .setHeader("Content-Type", constant("application/json"))
+                .setHeader("Platform-TenantId", constant(tenantId))
+                .setHeader(Exchange.HTTP_RAW_QUERY,
+                        simple("by=" + FILTER_BY_ERROR_CODE + "&value=${exchangeProperty." + ERROR_CODE + "}"))
+                .toD(getFilterUrl())
+                .log(LoggingLevel.INFO, "Status: ${header.CamelHttpResponseCode}")
                 .log(LoggingLevel.INFO, "Operations response: \n\n.. ${body}")
                 .to("direct:filter-response-handler");
 
         from("direct:get-recoverable-error-codes")
                 .id("get-recoverable-error-codes")
                 .log(LoggingLevel.INFO, "### Starting GET-RECOVERABLE-CODES route")
-                .toD(getFilterUrl(FILTER_BY_RECOVERABLE, true))
+                .removeHeader("*")
+                .removeHeader("Authorization")
+                .setHeader(Exchange.HTTP_METHOD, constant("GET"))
+                .setHeader("Content-Type", constant("application/json"))
+                .setHeader("Platform-TenantId", constant(tenantId))
+                .setHeader(Exchange.HTTP_RAW_QUERY,
+                        simple("by=" + FILTER_BY_ERROR_CODE + "&value=${exchangeProperty." + ERROR_CODE + "}"))
+                .toD(getFilterUrl())
                 .log(LoggingLevel.INFO, "Operations response: \n\n.. ${body}")
                 .to("direct:filter-response-handler");
 
         from("direct:get-non-recoverable-error-codes")
                 .id("get-non-recoverable-error-codes")
                 .log(LoggingLevel.INFO, "### Starting GET-NON-RECOVERABLE-CODES route")
-                .toD(getFilterUrl(FILTER_BY_RECOVERABLE, false))
+                .removeHeader("*")
+                .removeHeader("Authorization")
+                .setHeader(Exchange.HTTP_METHOD, constant("GET"))
+                .setHeader("Content-Type", constant("application/json"))
+                .setHeader("Platform-TenantId", constant(tenantId))
+                .setHeader(Exchange.HTTP_RAW_QUERY,
+                        simple("by=" + FILTER_BY_ERROR_CODE + "&value=${exchangeProperty." + ERROR_CODE + "}"))
+                .toD(getFilterUrl())
                 .log(LoggingLevel.INFO, "Operations response: \n\n.. ${body}")
                 .to("direct:filter-response-handler");
 
         from("direct:filter-response-handler")
```

Comment:
Are we doing any error handling for api calls to operations-app?

Reply:
no, we are expecting the operations-app endpoint for recoverable error code to be 100% available.

Reply:
done